pkg/controller/daemonset/daemonset_util.go

```diff
@@ -254,7 +254,10 @@ func unavailableCount(ds *appsv1alpha1.DaemonSet, numberToSchedule int) (int, er
 	}
 	r := ds.Spec.UpdateStrategy.RollingUpdate
 	if r == nil {
-		return 0, nil
+		return 1, nil
+	}
+	if r.MaxUnavailable == nil {
+		return 1, nil
```

Comment:
It should return `0` which is same as maxSurge ([here](https://github.com/openkruise/kruise/blob/master/pkg/controller/daemonset/daemonset_util.go#L266)), and then controller will set `maxUnavailable` to `1` if they are all `0` ([here](https://github.com/openkruise/kruise/blob/master/pkg/controller/daemonset/daemonset_update.go#L258)).